strawberry/relay/types.py

```diff
@@ -220,14 +220,22 @@ def resolve_type(self, info: Info) -> Type[Node]:
 
         """
         type_def = info.schema.get_type_by_name(self.type_name)
-        assert isinstance(type_def, StrawberryObjectDefinition)
+        if not isinstance(type_def, StrawberryObjectDefinition):
+            raise GlobalIDValueError(
+                f"Cannot resolve. GlobalID requires a GraphQL type, "
+                f"received {self.type_name}."
+            )
 
         origin = (
             type_def.origin.resolve_type
             if isinstance(type_def.origin, LazyType)
             else type_def.origin
         )
-        assert issubclass(origin, Node)
+        if not issubclass(origin, Node):
+            raise GlobalIDValueError(
+                f"Cannot resolve. GlobalID requires a GraphQL Node type, "
+                f"received {self.type_name}."
```

Comment:
```suggestion
                f"received `{self.type_name}`."
```

Reply:
@bellini666 how does this look? 😊

I wonder if there's people using. GlobalID without the `Node` interface

Reply:
> I wonder if there's people using. GlobalID without the Node interface

I doubt it. But even if they are, I'm not removing any of that functionality here, I'm just changing `AssertionError` being thrown to `GlobalIDValueError` with a better error message.

Reply:
This seems totally fine, I like the changes here :)